hudi-common/src/main/java/org/apache/hudi/common/engine/HoodieReaderContext.java

```diff
@@ -285,6 +277,14 @@ public abstract HoodieRecord<T> constructHoodieRecord(Option<T> recordOption,
    */
   public abstract T seal(T record);
 
+  /**
+   * Convert engine specific row into binary format.
+   *
+   * @param record The engine row
+   * @return row with binary format
+   */
+  public abstract T toBinaryRow(T record);
```

Comment:
Can you add some context for why this is required?

Reply:
Instead of enforcing all implementations to have a concept of a Binary version, can we move this into the code that reads the data into format `T`? Or possible move it to the `seal` functionality?

Reply:
We think through this and think should add a ne interface for 2 reasons:

- `seal` is only for copy purposes, not care about the row format itself;
- `toBinaryRow` will just focus on row format transformation.

Thus to avoid unnecessary copy or transf of rows.

Reply:
We should only enforce the row that would be put in spillable map has binary format. And `seal` is called in multiple places, e.g., `baseRecord` read from base file is also sealed, where `toBinary` is not necessary to avoid additional costs. That's the context we introduce a new method `toBinary` to separate `seal`.

Reply:
We can move this conversion directly to the code that converts from the avro to the row. This doesn't need to be part of the abstract class in my opinion.

Reply:
We can also make this the responsibility of the merger though to return an optimal representation. If the merger returns an avro, we will run it through the same conversion code used when reading the log files